### PR TITLE
Fragmented trajectory support

### PR DESCRIPTION
**Pull request by clonker**
This PR deals with issue #572. It also adds a skip parameter for the readers such that the first few frames can be skipped without using lag.

### CODE REVIEW

**Comment by franknoe** [comment]
Thank you! Let's wait with merging this for 2.1

Am 05/11/15 um 13:04 schrieb Moritz Hoffmann:

> This PR deals with issue #572 
> https://github.com/markovmodel/PyEMMA/issues/572. It also adds a 
> skip parameter for the readers such that the first few frames can be 
> skipped without using lag.
> 
> ---
> 
> ```
>     You can view, comment on, or merge this pull request online at:
> ```
> 
> https://github.com/markovmodel/PyEMMA/pull/610
> 
> ```
>     Commit Summary
> ```
> - [coordinates/util/patches] change iterload generator to iterator
>   so dynamic settings like chunksize can be maintained
> - [coordinates/data/readers] add skip parameter such that the first
>   few frames can be skipped without using lag
> - [coordinates/data/fragmented_trajectory_reader] Apidoc adjustments
>   and fragmented trajectory support
>   
>   ```
>   File Changes
>   ```
> - _M_ pyemma/coordinates/api.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-0 (17)
> - _M_ pyemma/coordinates/data/data_in_memory.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-1 (15)
> - _M_ pyemma/coordinates/data/feature_reader.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-2 (15)
> - _A_ pyemma/coordinates/data/fragmented_trajectory_reader.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-3 (250)
> - _M_ pyemma/coordinates/data/interface.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-4 (17)
> - _M_ pyemma/coordinates/data/numpy_filereader.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-5 (15)
> - _M_ pyemma/coordinates/data/py_csv_reader.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-6 (4)
> - _M_ pyemma/coordinates/data/util/reader_utils.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-7 (10)
> - _M_ pyemma/coordinates/tests/test_csvreader.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-8 (35)
> - _M_ pyemma/coordinates/tests/test_datainmemory.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-9 (28)
> - _M_ pyemma/coordinates/tests/test_featurereader.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-10 (25)
> - _A_ pyemma/coordinates/tests/test_fragmented_trajectory.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-11 (138)
> - _M_ pyemma/coordinates/tests/test_numpyfilereader.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-12 (25)
> - _M_ pyemma/coordinates/util/patches.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-13 (242)
>   
>   ```
>   Patch Links:
>   ```
> - https://github.com/markovmodel/PyEMMA/pull/610.patch
> - https://github.com/markovmodel/PyEMMA/pull/610.diff
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/markovmodel/PyEMMA/pull/610.

## 

---

Prof. Dr. Frank Noe
Head of Computational Molecular Biology group
Freie Universitaet Berlin

Phone: (+49) (0)30 838 75354
Web:   research.franknoe.de

## Mail:  Arnimallee 6, 14195 Berlin, Germany

**Comment by franknoe** [comment]
I meant: wait until 2.0.2. is released, i.e. it will go into 2.1

Am 05/11/15 um 13:04 schrieb Moritz Hoffmann:

> This PR deals with issue #572 
> https://github.com/markovmodel/PyEMMA/issues/572. It also adds a 
> skip parameter for the readers such that the first few frames can be 
> skipped without using lag.
> 
> ---
> 
> ```
>     You can view, comment on, or merge this pull request online at:
> ```
> 
> https://github.com/markovmodel/PyEMMA/pull/610
> 
> ```
>     Commit Summary
> ```
> - [coordinates/util/patches] change iterload generator to iterator
>   so dynamic settings like chunksize can be maintained
> - [coordinates/data/readers] add skip parameter such that the first
>   few frames can be skipped without using lag
> - [coordinates/data/fragmented_trajectory_reader] Apidoc adjustments
>   and fragmented trajectory support
>   
>   ```
>   File Changes
>   ```
> - _M_ pyemma/coordinates/api.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-0 (17)
> - _M_ pyemma/coordinates/data/data_in_memory.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-1 (15)
> - _M_ pyemma/coordinates/data/feature_reader.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-2 (15)
> - _A_ pyemma/coordinates/data/fragmented_trajectory_reader.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-3 (250)
> - _M_ pyemma/coordinates/data/interface.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-4 (17)
> - _M_ pyemma/coordinates/data/numpy_filereader.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-5 (15)
> - _M_ pyemma/coordinates/data/py_csv_reader.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-6 (4)
> - _M_ pyemma/coordinates/data/util/reader_utils.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-7 (10)
> - _M_ pyemma/coordinates/tests/test_csvreader.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-8 (35)
> - _M_ pyemma/coordinates/tests/test_datainmemory.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-9 (28)
> - _M_ pyemma/coordinates/tests/test_featurereader.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-10 (25)
> - _A_ pyemma/coordinates/tests/test_fragmented_trajectory.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-11 (138)
> - _M_ pyemma/coordinates/tests/test_numpyfilereader.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-12 (25)
> - _M_ pyemma/coordinates/util/patches.py
>   https://github.com/markovmodel/PyEMMA/pull/610/files#diff-13 (242)
>   
>   ```
>   Patch Links:
>   ```
> - https://github.com/markovmodel/PyEMMA/pull/610.patch
> - https://github.com/markovmodel/PyEMMA/pull/610.diff
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/markovmodel/PyEMMA/pull/610.

## 

---

Prof. Dr. Frank Noe
Head of Computational Molecular Biology group
Freie Universitaet Berlin

Phone: (+49) (0)30 838 75354
Web:   research.franknoe.de

## Mail:  Arnimallee 6, 14195 Berlin, Germany

**Comment by marscher** [comment]
Cool! Thank you! This involved a lot of changes...

File: pyemma/coordinates/data/fragmented_trajectory_reader.py [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+    def __iter__(self):
+        return self
+
+    @property
+    def ctx(self):
+        return self._ctx
+
+    @ctx.setter
+    def ctx(self, value):
+        ctx_copy = copy(value)
```
Comment (marscher) [inline pyemma/coordinates/data/fragmented_trajectory_reader.py:41]: intermediate code (O'BACHT Seiteneffekte!)

**Comment by marscher** [comment]
File "c:\miniconda\envs_test\lib\site-packages\pyemma-2.0.2+11.g9668c9c-py2.7-win32.egg\pyemma\coordinates\tests\test_stride.py", line 121, in tearDownClass
    os.unlink(fname)
WindowsError: [Error 32] The process cannot access the file because it is being used by another process: 'c:\users\appveyor\appdata\local\temp\1\tmpzmiwsz.xtc'

**Comment by marscher** [comment]
The error message suggests that you're not closing the filehandle in the reader.

**Comment by marscher** [comment]
included in branch "devel_refactor_transformer"

**Pull request closed without merging**